annotation/app/annotations/main.py

```diff
@@ -60,14 +66,35 @@ def row_to_dict(row) -> dict:
     }
 
 
+class NotConfiguredException(Exception):
+    pass
+
+
 def connect_s3(bucket_name: str) -> boto3.resource:
-    s3_resource = boto3.resource(
-        "s3",
-        endpoint_url=ENDPOINT_URL,
-        aws_access_key_id=AWS_ACCESS_KEY_ID,
-        aws_secret_access_key=AWS_SECRET_ACCESS_KEY,
-    )
+    boto3_config = {}
+    if S3_CREDENTIALS_PROVIDER == "minio":
+        logger_.info(f"S3_Credentials provider - {S3_CREDENTIALS_PROVIDER}")
+        boto3_config.update(
+            {
+                "aws_access_key_id": AWS_ACCESS_KEY_ID,
+                "aws_secret_access_key": AWS_SECRET_ACCESS_KEY,
+                "endpoint_url": ENDPOINT_URL,
+            }
+        )
+    elif S3_CREDENTIALS_PROVIDER == "aws_iam":
+        logger_.info(f"S3_Credentials provider - {S3_CREDENTIALS_PROVIDER}")
```

Comment:
What about moving logging to a single line (L92)?
And change its logging level to `debug`?

Reply:
agreed